### DISCUSSION TITLE
Current State

### DISCUSSION THREAD

**Discussion by DirkLewis** [drseg/swift-fsm#3] [General]
Started looking and this little gem and I seem to be missing how to get the current state of the machine.  Looks like your test crack open for a deeper look but if I am just writing tests to confirm transitions I can not seem to get the current.
thanks in advance

**Comment by drseg** [comment]
I answered this in your other post, however just as a further note you'll see that when in a debug environment it also logs transitions to the console.

Eg:

`[23:15:34 SwiftFSM] Info: transition { define(idle) | when(didTapSettings) | then(presentingSettings) } was executed`
